on_device_training/training_api_demo/mnist_training_example.ipynb

```diff
@@ -0,0 +1,437 @@
+    "\n",
+    "    def forward(self, model_input):\n",
+    "        out = self.fc1(model_input)\n",
+    "        out = self.relu(out)\n",
+    "        out = self.fc2(out)\n",
+    "        return out\n",
+    "\n",
+    "# Create a Simple instance.\n",
+    "device = \"cpu\"\n",
+    "batch_size, input_size, hidden_size, output_size = 64, 784, 500, 10\n",
+    "pt_model = MNISTNet(input_size, hidden_size, output_size).to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generating forward only graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate a random input.\n",
+    "model_inputs = (torch.randn(batch_size, input_size, device=device),)\n",
+    "\n",
+    "model_outputs = pt_model(*model_inputs)\n",
+    "if isinstance(model_outputs, torch.Tensor):\n",
+    "    model_outputs = [model_outputs]\n",
+    "    \n",
+    "dynamic_axes = {}\n",
+    "input_names = []\n",
+    "output_names = []\n",
+    "\n",
+    "for i, model_input in enumerate(model_inputs):\n",
+    "    input_name = f\"input-{i}\"\n",
+    "    input_names.append(input_name)\n",
+    "    dynamic_axes[input_name] = {}\n",
+    "    for dim_idx in range(len(model_input.shape)):\n",
+    "        dynamic_axes[input_name].update({dim_idx: f\"{input_name}_dim{dim_idx}\"})\n",
+    "\n",
+    "for i, model_output in enumerate(model_outputs):\n",
+    "    output_name = f\"output-{i}\"\n",
+    "    output_names.append(output_name)\n",
+    "    dynamic_axes[output_name] = {}\n",
+    "    for dim_idx in range(len(model_output.shape)):\n",
+    "        dynamic_axes[output_name].update({dim_idx: f\"{output_name}_dim{dim_idx}\"})\n",
+    "\n",
+    "f = io.BytesIO()\n",
+    "torch.onnx.export(\n",
+    "    pt_model,\n",
+    "    model_inputs,\n",
+    "    f,\n",
+    "    input_names=input_names,\n",
+    "    output_names=output_names,\n",
+    "    opset_version=14,\n",
+    "    do_constant_folding=False,\n",
+    "    training=torch.onnx.TrainingMode.TRAINING,\n",
+    "    dynamic_axes=dynamic_axes,\n",
+    "    export_params=True,\n",
+    "    keep_initializers_as_inputs=False,\n",
+    ")\n",
+    "onnx_model = onnx.load_model_from_string(f.getvalue())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### After creating forward only graph, we can now create the training graph.\n",
+    "\n",
+    "The first step is creating a simple class that inherits from onnxblock.TrainingModel, and define the loss function.\n",
+    "the build function defines the output of our model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Creating a class with a Loss function.\n",
+    "class SimpleModelWithCrossEntropyLoss(onnxblock.TrainingModel):\n",
```

Comment:
Can we call this something else as well? Maybe `ModelWithLoss`, to abstract away the contents of the model?